core/chains/evm/legacy.go

```diff
@@ -110,43 +109,38 @@ func ClobberDBFromEnv(db *sqlx.DB, config LegacyEthNodeConfig, lggr logger.Logge
 	return nil
 }
 
-// SetupMultiplePrimaries is a hack/shim method to allow node operators to
-// specify multiple nodes via ENV
+// SetupNodes is a hack/shim method to allow node operators to specify multiple nodes via ENV.
 // See: https://app.shortcut.com/chainlinklabs/epic/33587/overhaul-config?cf_workflow=500000005&ct_workflow=all
-func SetupMultiplePrimaries(db *sqlx.DB, cfg LegacyEthNodeConfig, lggr logger.Logger) (err error) {
+func SetupNodes(db *sqlx.DB, cfg LegacyEthNodeConfig, lggr logger.Logger) (err error) {
 	if cfg.EthereumNodes() == "" {
 		return nil
 	}
 
-	lggr.Info("EVM_NODES was set; clobbering evm_nodes table")
-	_, err = db.Exec(`TRUNCATE evm_nodes;`)
-	if err != nil {
-		return errors.Wrap(err, "failed to truncate evm_nodes table while inserting nodes set by EVM_NODES")
-	}
-
 	var nodes []evmtypes.Node
 	if err = json.Unmarshal([]byte(cfg.EthereumNodes()), &nodes); err != nil {
-		return errors.Wrapf(err, "invalid nodes json, got: %q", cfg.EthereumNodes())
+		return errors.Wrapf(err, "invalid EVM_NODES json, got: %q", cfg.EthereumNodes())
```

Comment:
Do we need to check if EVM_NODES has empty array?

Reply:
Hmmmm I suppose that could improve UX. Does empty array suggest that the user made a mistake, and we should error out as invalid?

Reply:
Yes that's what I am thinking of